feature-libs/order/components/order-details/order-details-extended/consignment-tracking/consignment-tracking-link.component.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  inject,
+  OnDestroy,
+  OnInit,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import {
+  LaunchDialogService,
+  LAUNCH_CALLER,
+  OutletContextData,
+} from '@spartacus/storefront';
+import { of, Subscription } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { Consignment } from '@spartacus/order/root';
+import { OrderConsignmentsService } from '../../order-consignments.service';
+
+@Component({
+  selector: 'cx-consignment-tracking-link',
+  templateUrl: './consignment-tracking-link.component.html',
+  host: { class: 'cx-list-header col-12' },
+})
+export class ConsignmentTrackingLinkComponent implements OnInit, OnDestroy {
+  consignment: Consignment;
+  protected subscription = new Subscription();
+  protected orderConsignmentsService = inject(OrderConsignmentsService);
+  protected launchDialogService = inject(LaunchDialogService);
+  protected vcr = inject(ViewContainerRef);
+  protected cd = inject(ChangeDetectorRef);
+  protected outlet = inject(OutletContextData);
+
+  @ViewChild('element') element: ElementRef;
```

Comment:
[QUESTION]
I have a problem to find the declaration of `#element` identifier in the component's template. Could you help me understand how it works, please? Perhaps I'm missing something obvious.

Reply:
Created a dummy property to just invoke launchDialogService.openDialog. But since it is an optional parameter, I have now replaced this with undefined.